mayastor/src/rebuild.rs

```diff
@@ -191,80 +224,115 @@ impl RebuildJob {
             destination_hdl,
             start,
             end,
-            current: start,
+            next: start,
             block_size,
             segment_size_blks,
-            copy_buffer,
+            tasks,
             complete_fn,
             complete_chan: unbounded::<RebuildState>(),
             state: RebuildState::Pending,
         })
     }
 
-    /// Rebuilds a non-healthy child from a healthy child from start to end
+    // Runs the management async task that kicks off N rebuild copy tasks and
+    // awaits each completion before kicking off more tasks until the
+    // bdev is fully rebuilt
```

Comment:
I wonder if we should reword this a bit "...awaits each completion before kicking off more tasks...". I'm not sure if it's clear whether we kick off another task as one finishes or if this suggests we wait for all to complete before kicking off more tasks.

Reply:
hmm yeah...
"awaits completions and kicks off more tasks after each completion" ?

Reply:
> // Runs the management async task that kicks off N rebuild copy tasks and
> // awaits each completion. When any task completes it kicks off another until the
> // bdev is fully rebuilt